nexus-sdk/src/main/java/io/nexusrpc/OperationHandler.java

```diff
@@ -0,0 +1,228 @@
+package io.nexusrpc;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.function.BiFunction;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Handler for an operation.
+ *
+ * <p>An instance of this must be returned from an {@link OperationImpl} annotated method in a
+ * {@link ServiceImpl}. Simple synchronous operations can use {@link #sync}, more complex or
+ * asynchronous operations can manually implement this interface.
+ *
+ * @param <T> The parameter type of the operation. This can be {@link Void} for no parameter.
+ * @param <R> the return type of the operation. This can be {@link Void} for no return.
+ */
+public interface OperationHandler<T, R> {
+  /**
+   * Create an operation handler for a synchronous operation backed by the given function. This
+   * function will be called for each operation invocation. All other calls to the operation are not
+   * supported.
+   */
+  static <T, R> OperationHandler<T, R> sync(BiFunction<StartContext, @Nullable T, R> func) {
+    return new OperationHandler<T, R>() {
+      @Override
+      public StartResult<R> start(StartContext ctx, @Nullable T param) {
+        return StartResult.sync(func.apply(ctx, param));
+      }
+
+      @Override
+      public R fetchResult(FetchResultContext ctx) {
+        throw new UnsupportedOperationException("Not supported on sync operation");
+      }
+
+      @Override
+      public OperationInfo fetchInfo(FetchInfoContext ctx) {
+        throw new UnsupportedOperationException("Not supported on sync operation");
+      }
+
+      @Override
+      public void cancel(CancelContext ctx) {
+        throw new UnsupportedOperationException("Not supported on sync operation");
+      }
+    };
+  }
+
+  /**
+   * Handle the start of an operation.
+   *
+   * <p>The operation can be synchronous or asynchronous. Synchronous operations can return the
+   * result via {@link StartResult#sync}. Asynchronous operations can return the started operation
+   * ID via {@link StartResult#async}.
+   *
+   * @param ctx Context containing details about the call.
+   * @param param Parameter for the operation. This may be null if the parameter was not given.
+   * @return The start result which is a synchronous value or an operation ID representing an
+   *     asynchronously running operation.
+   * @throws OperationUnsuccessfulException If thrown, can have failure details and state such as
+   *     saying the operation was cancelled.
+   * @throws RuntimeException Any other exception fails the operation start as an operation failure.
+   */
+  StartResult<R> start(StartContext ctx, @Nullable T param) throws OperationUnsuccessfulException;
```

Comment:
In Go we called this `HandlerStartResult` since the client also has a `StartResult` that's slightly different. I would rename in prep for a client implementation.

https://pkg.go.dev/github.com/nexus-rpc/sdk-go@v0.0.9/nexus#ClientStartOperationResult
https://pkg.go.dev/github.com/nexus-rpc/sdk-go@v0.0.9/nexus#HandlerStartOperationResult

Reply:
`StartResult` is an inner class, it won't get confused with some other thing. This is one of the reasons I made them inner classes.

Reply:
Ah nice